## <center>Практическое задание к лекции 3.

**1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):**  
   **а) среднее арифметическое,**  
   **б) среднее квадратичное отклонение,**  
   **с) смещенную и несмещенную оценки дисперсий для данной выборки.**

In [21]:
import numpy as np
from math import factorial

In [2]:
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

a) Расчитаем среднее арифметическое по формуле: $\displaystyle M(X)=\frac{1}{n}\sum^n_{i=1}x_i$

In [3]:
def mean_value(array):
    return sum(array)/len(array)

In [4]:
print(f'Среднее арифметическое для данной выборки М(Х) = {mean_value(arr): .2f}')

Среднее арифметическое для данной выборки М(Х) =  65.30


In [5]:
np.mean(arr)

65.3

б) Расчитаем среднее квадратичное отклонение выборки по формуле: $\sigma=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}}$

In [12]:
def mean_square_deviation(array):
    square_dev=(array-mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)

In [19]:
print(f'Среднее квадратичное отклонение для данной выборки SD = {mean_square_deviation(arr): .4f}')

Среднее квадратичное отклонение для данной выборки SD =  30.8239


Проверим с помощью функции std:  

In [14]:
np.std(arr)

30.823854398825596

в) Расчитаем смещенную оценку дисперсии по формуле: $\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}$  
а несмещённую по формуле: $\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}$

In [15]:
def dispers(array, no_off=False):
#аргумент no_off отвечает за расчет смещенной, или несмещенной оценки дисперсии. По умолчанио расчитывается смещенная.#
    square_dev=(array-mean_value(array))**2
    return sum(square_dev)/(len(square_dev)-1) if no_off else sum(square_dev)/len(square_dev)

In [20]:
print(f'Смещенная оценка дисперсии для данной выборки D = {dispers(arr): .4f}\n'
      f'Немещенная оценка дисперсии для данной выборки D = {dispers(arr, True): .4f}'
     )

Смещенная оценка дисперсии для данной выборки D =  950.1100
Немещенная оценка дисперсии для данной выборки D =  1000.1158


Проверим с помощью функции var:

In [17]:
np.var(arr)

950.11

In [18]:
np.var(arr, ddof=1)

1000.1157894736842

---  
#### Ответ:  
а) $М(Х)= 65.30$;  
б) $\sigma = 30.8239$;  
в) $\sigma^2_{\text{смещ}}=950.11$, $\sigma^2_{\text{несмещ}}=1000.1158$.  

---

**2. В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?**

Пусть $A$-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда это событие может произойти следующими способами:  
   а) $A_1$ из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;  
   б) $A_2$ из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;  
   в) $A_3$ из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.  
Вероятность события $A$ будет определяться как сумма вероятностей этих событий.  
Определим вероятность события $A_1$:  
$P(A_1)=\frac{C^2_3}{C^2_8}\cdot\frac{C^3_5\cdot C^1_7}{C^4_{12}}$

In [22]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

In [31]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(A1) = {P1: .4f}'
     )

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) =  0.0152


Определим вероятность события $A_2$:  
$P(A_2)=\frac{C^1_5\cdot C^1_3}{C^2_8}\cdot\frac{C^2_5\cdot C^2_7}{C^4_{12}}$

In [32]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(A2) = {P2: .4f}'
     )

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) =  0.2273


Определим вероятность события $A_3$:  
$P(A_3)=\frac{C^2_5}{C^2_8}\cdot\frac{C^1_5\cdot C^3_7}{C^4_{12}}$

In [34]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(A2) = {P3: .4f}'
     )

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A2) =  0.1263


Вычислим $P(A)=P(A_1+A_2+A_3)=P(A_1)+P(A_2)+P(A_3)$

In [35]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


---  
#### Ответ:  
Вероятность того, что 3 мяча белые $P(A)\approx 0.3687\approx 36.87\%$  

---

**3. На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен:**  
   **a) первым спортсменом;**  
   **б) вторым спортсменом;**  
   **в) третьим спортсменом.**

Рассмотрим следующие события:  
$A$-в результате выстрела, цель поражена;  
$B_1$ выстрел произведен первым спортсменом;  
$B_2$ выстрел произведен вторым спортсменом;  
$B_3$ выстрел произведен третьим спортсменом.  
Вероятности того, что выстрел произведен определенным спортсменом равны, т.к. выстрел 1, и произвести его могут только 3 этих спортсмена, т.е. события $B_1, B_2, B_3$ составляют полную группу событий.  
$\Rightarrow P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$,  
Вероятность того, что произошло событие $A$, при условии наступления события $B_n$ даны в условии задачи, т.е. $P_{B_1}(A)=0.9, P_{B_2}(A)=0.8, P_{B_3}(A)=0.6$, и, по формуле **полной** вероятности, вероятность наступления события $A$ - $P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$  

In [36]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


Теперь, исходя из того, что событие $A$ уже произошло, с помощью формулы **Байеса** оценим вероятность того, что имело место одно из событий $B_n$  
$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}\Rightarrow\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}\\P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$  


In [38]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .4f}.'
     )

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.260870.


---
#### Ответ:  
а) Вероятность того, что выстрел произвёл первый спортсмен:  $P_{A}(B_1)\approx0.3913\approx39.13\%$;  
б) Вероятность того, что выстрел произвёл второй спортсмен:  $P_{A}(B_2)\approx0.3478\approx34.78\%$;  
в) Вероятность того, что выстрел произвёл третий спортсмен:  $P_{A}(B_3)\approx0.2609\approx26.09\%$.    

---

**4. В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится:**  
   **a) на факультете A;**  
   **б) на факультете B;**    
   **в) на факультете C?**  

Пусть $n$ - количество студентов на факультете $A$, тогда на факультете $B$ учится $n$ студентов, а на факультете $C$ $2n$ студентов.  
Введем события $S_A, S_B, S_C$, что сессию сдаёт студент факультетов $A$, $B$, и $С$ соответственно, и событие $D$-сессия сдана.  
события $S_A$, $S_B$, и $S_С$ образуют полную группу событий, $\Rightarrow P(S_A)+P(S_B)+P(S_C)=1$  
T.e. $\frac{1}{n}+\frac{1}{n}+\frac{1}{2n}=1\\n=2.5\\\Rightarrow P(S_A)=\frac{1}{n}=0.25, P(S_B)=\frac{1}{n}=0.25, P(S_C)=\frac{1}{2n}=0.5$  
Условные вероятности Наступления события $D$, при условии что произошло одно из событий $S_n$ даны в условии задачи.  
Полная вероятность наступления события $D$ исходя из формулы **полной** вероятности:  
$P(D)=0.25\cdot 0.8+0.25\cdot 0.7+0.5\cdot 0.9$  

In [40]:
PD=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события D, P(D) = {PD}.')

Полная вероятность наступления события D, P(D) = 0.825.


По формуле **Байеса** вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию.  
$P_D(S_A)=\frac{P(S_A)\cdot P_{S_A}(D)}{P(D)}$ студент учится на факультете А  
$P_D(S_B)=\frac{P(S_B)\cdot P_{S_B}(D)}{P(D)}$ студент учится на факультете В  
$P_D(S_C)=\frac{P(S_C)\cdot P_{S_C}(D)}{P(D)}$ студент учится на факультете С

In [43]:
PDSA=0.25*0.8/PD
PDSB=0.25*0.7/PD
PDSC=0.5*0.9/PD
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.'
      )

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


---  
#### Ответ:  
Вероятность того, что студент учится на факультете А:  $P_D(S_A)\approx0.2424\approx24.24\%$;  
Вероятность того, что студент учится на факультете B:  $P_D(S_B)\approx0.2121\approx21.21\%$;  
Вероятность того, что студент учится на факультете C:  $P_D(S_C)\approx0.5455\approx54.55\%$.  

---

**5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:**  
   **а) все детали;**  
   **б) только две детали;**  
   **в) хотя бы одна деталь;**  
   **г) от одной до двух деталей?**

Введем следующие события:  
   а) Событие $A_1$-вышла из строя первая деталь, по условию задачи-вероятность этого события: $P(A_1)=0.1$;  
   б) Событие $\bar A_1$-противоположное событию $A_1$ - первая деталь не вышла из строя, вероятность этого события: $P(\bar A_1)=1-P(A_1)=1-0.1=0,9$;  
   в) Событие $A_2$-вышла из строя вторая деталь, по условию задачи-вероятность этого события: $P(A_2)=0.2$;  
   г)  Событие $\bar A_2$-противоположное событию $A_2$ - вторая деталь не вышла из строя, вероятность этого события: $P(\bar A_2)=1-P(A_2)=1-0.2=0,8$;  
   д) Событие $A_3$-вышла из строя первая деталь, по условию задачи-вероятность этого события: $P(A_3)=0.25$;  
   е) Событие $\bar A_3$-противоположное событию $A_3$ - третья деталь не вышла из строя, вероятность этого события: $P(\bar A_3)=1-P(A_3)=1-0.25=0,9$;  
   

---  
а) Для того, чтобы произошло событие "вышли из строя все детали", необходимо, чтобы **одновременно** произошли события $A_1$, $A_2$, и $A_3$. А вероятностью этого события $P(3)$ будет произведением вероятностей этих событий.  
$P(3)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$

In [5]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


---  
б) Для того, чтобы произошло событие  "вышли из строя 2 детали", необходимо что бы случилось **одна из** следующих комбинаций событий:  
   1. Произошли события $A_1$ **и** $A_2$ **и** $\bar A_3$;  
   2. Произошли события $A_1$ **и** $A_3$ **и** $\bar A_2$;  
   3. Произошли события $A_2$ **и** $A_3$ **и** $\bar A_1$.  
   
А вероятностью этого события  $P(2)$  будет **суммой** вероятностей этих комбинаций событий.  
$P(2)=P(A_1)\cdot P(A_2)\cdot P(\bar A_3)+P(A_1)\cdot P(A_3)\cdot P(\bar A_2)+P(A_2)\cdot P(A_3)\cdot P(\bar A_1)=0.1\cdot 0.2\cdot 0.75+0.1\cdot 0.25\cdot 0.8+0.2\cdot 0.25\cdot 0.9$

In [7]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут только 2 детали Р(2) =  0.0800


---  
в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как противоположное событию "не вышло из строя ни одной детали".   
Пусть $P(0)$ вероятность события "не вышло из строя ни одной детали". Данное событие может произойти при одновременном наступлении событий $\bar A_1$, $\bar A_2$, и $\bar A_3$.  
$P(0)=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$

In [9]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


Тогда вероятность события "выйдет из строя хотя бы одна деталь" $P(\ge 1)=1-P(0)$

In [10]:
P_0=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P_0: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) =  0.4600


---  
г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" **или** из строя выйдут 2 детали, следовательно его вероятность будет равна сумме вероятностей этих событий: $P(1,2)=P(1)+P(2)$  
$P(2)=0.08$, найдем $P(1)$:  
  
<u>**Способ 1:**</u>  
  
Для того, чтобы произошло событие  "из строя выйдет одна деталь", необходимо что бы случилось **одна из** следующих комбинаций событий:  
   1. Произошли события $A_1$ **и** $\bar A_2$ **и** $\bar A_3$;  
   2. Произошли события $A_2$ **и** $\bar A_1$ **и** $\bar A_3$;  
   3. Произошли события $A_3$ **и** $\bar A_1$ **и** $\bar A_2$.  
   
А вероятностью этого события  $P(1)$  будет **суммой** вероятностей этих комбинаций событий.  
$P(1)=P(A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)+P(A_2)\cdot P(\bar A_1)\cdot P(\bar A_3)+P(A_3)\cdot P(\bar A_1)\cdot P(\bar A_2)=0.1\cdot 0.8\cdot 0.75+0.2\cdot 0.9\cdot 0.75+0.25\cdot 0.9\cdot 0.8$  
---

In [15]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


<u>**Способ 2:**</u>  
  
События "не выйдет из строя ни одной детали", "из строя выйдет одна деталь", "из строя выйдет 2 детали", "из строя выйдут все детали" - **несовместны**, и образуют полную группу событий, поэтому сумма вероятностей наступления этих событий равна $1$.  
$P(0)+P(1)+P(2)+P(3)=1\\0.54+P(1)+0.08+0.005=1\\P(1)=0.375$  
---

$P(1,2)=P(1)+P(2)=0.375+0.08=0.455$  
  
---  
#### Ответ:  
а)  Вероятность того что в первый месяц выйдут из строя все детали $P(3)=0.005=0.50\%$  
б)  Вероятность того что в первый месяц выйдут из строя 2 детали $P(2)=0.08=8.00\%$  
в)  Вероятность того что в первый месяц выйдут из строя хотя бы одна деталь $P(\ge 1)=0.46=46.00\%$  
г)  Вероятность того что в первый месяц выйдут из строя от одной до двух деталей $P(1,2)=0.455=45.50\%$  

---